src/uu/du/src/du.rs

```diff
@@ -443,8 +480,77 @@ impl UError for DuError {
             Self::InvalidMaxDepthArg(_)
             | Self::SummarizeDepthConflict(_)
             | Self::InvalidTimeStyleArg(_)
-            | Self::InvalidTimeArg(_) => 1,
+            | Self::InvalidTimeArg(_)
+            | Self::InvalidGlob(_) => 1,
+        }
+    }
+}
+
+// Read a file and return each line in a vector of String
+fn file_as_vec(filename: impl AsRef<Path>) -> Vec<String> {
+    let file = File::open(filename).expect("no such file");
+    let buf = BufReader::new(file);
+
+    buf.lines()
+        .map(|l| l.expect("Could not parse line"))
+        .collect()
+}
+
+// Given the --exclude-from and/or --exclude arguments, returns the globset lists
+// to ignore the files
+fn get_globset_ignore(matches: &ArgMatches) -> UResult<GlobSet> {
+    let mut excludes_from = if matches.is_present(options::EXCLUDE_FROM) {
+        match matches.values_of(options::EXCLUDE_FROM) {
+            Some(all_files) => {
+                let mut exclusion = Vec::<String>::new();
+                // Read the exclude lists from all the files
+                // and add them into a vector of string
+                let files: Vec<String> = all_files.clone().map(|v| v.to_owned()).collect();
+                for f in files {
+                    exclusion.extend(file_as_vec(&f));
+                }
+                exclusion
+            }
+            None => Vec::<String>::new(),
+        }
+    } else {
+        Vec::<String>::new()
+    };
+
+    let mut excludes = if matches.is_present(options::EXCLUDE) {
+        match matches.values_of(options::EXCLUDE) {
+            Some(v) => {
+                // Read the various arguments
+                v.clone().map(|v| v.to_owned()).collect()
+            }
+            None => Vec::<String>::new(),
+        }
+    } else {
+        Vec::<String>::new()
+    };
+
+    // Merge the two lines
+    excludes.append(&mut excludes_from);
+    if !&excludes.is_empty() {
+        let mut builder = GlobSetBuilder::new();
+        // Create the globset of excludes
+        for f in excludes {
+            if matches.is_present(options::VERBOSE) {
+                println!("adding {:?} to the exclude list ", &f);
+            }
+            let g = match Glob::new(&f) {
+                Ok(glob) => glob,
+                Err(err) => return Err(DuError::InvalidGlob(err.to_string()).into()),
+            };
+            builder.add(g);
+        }
+        match builder.build() {
+            // Handle the error. Not sure when this happens
```

Comment:
This is described in https://docs.rs/globset/latest/globset/enum.ErrorKind.html. It fails on cases where for example a character class is not closed.